### PR TITLE
fix(NoCode): Fix product analytics

### PR DESCRIPTION
**Pull request by dexter-mh-lee**
Migrate hydrators to call the new EntityService. 
Fix index name for datahub usage event. 
Fix issue where the reindex gets triggered on each deploy. 

## Checklist
- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable)

### CODE REVIEW

File: gms/client/src/main/java/com/linkedin/entity/client/EntityClient.java [hunk 1]
```diff
@@ -24,240 +19,232 @@
...
+
+  /**
+   * Gets browse path(s) given dataset urn
+   *
+   * @param urn urn for the entity
+   * @return list of paths given urn
+   * @throws RemoteInvocationException
+   */
+  @Nonnull
+  public StringArray getBrowsePaths(@Nonnull Urn urn) throws RemoteInvocationException {
```
Comment (jjoyce0510) [inline gms/client/src/main/java/com/linkedin/entity/client/EntityClient.java:245]: This appears to be just formatting -- is that correct?

Comment (dexter-mh-lee) [reply gms/client/src/main/java/com/linkedin/entity/client/EntityClient.java:245]: Yeah :( Ran formatter for my code. and it ran for everything.

Comment (dexter-mh-lee) [reply gms/client/src/main/java/com/linkedin/entity/client/EntityClient.java:245]: The only logic is to take out the key aspect generation logic into the utils

File: metadata-io/src/main/java/com/linkedin/metadata/entity/EntityService.java [hunk 2]
```diff
@@ -131,11 +125,8 @@ public abstract RecordTemplate getAspect(
    * @param auditStamp an {@link AuditStamp} containing metadata about the writer & current time
    * @return the {@link RecordTemplate} representation of the written aspect object
    */
-  public abstract RecordTemplate ingestAspect(
-      @Nonnull final Urn urn,
-      @Nonnull final String aspectName,
-      @Nonnull final RecordTemplate newValue,
-      @Nonnull final AuditStamp auditStamp);
+  public abstract RecordTemplate ingestAspect(@Nonnull final Urn urn, @Nonnull final String aspectName,
```
Comment (jjoyce0510) [inline metadata-io/src/main/java/com/linkedin/metadata/entity/EntityService.java:128]: Why these changes? Is this checkstyle? I much prefer new lines for readability.

File: metadata-jobs/mae-consumer-job/build.gradle [hunk 3]
```diff
@@ -13,6 +13,7 @@ dependencies {
     avro project(path: ':metadata-models', configuration: 'avroSchema')
 
     compile project(':li-utils')
+    compile project(':gms:client')
```
Comment (jjoyce0510) [inline metadata-jobs/mae-consumer-job/build.gradle:16]: Once my merge PR gets in we may need to adjust this

File: metadata-io/src/main/java/com/linkedin/metadata/entity/EntityService.java (hunk 2 above)
Comment (dexter-mh-lee) [reply metadata-io/src/main/java/com/linkedin/metadata/entity/EntityService.java]: Tweaked the linkedin formatter to make it do new lines for this class. But a lot of other classes follow this style tho.

File: metadata-jobs/mae-consumer-job/src/main/java/com/linkedin/metadata/kafka/config/HydratorFactoryConfig.java [hunk 4]
```diff
@@ -22,6 +24,7 @@
   @Bean
   public HydratorFactory getHydratorFactory() {
     Client restClient = DefaultRestliClientFactory.getRestLiClient(gmsHost, gmsPort, gmsUseSSL, gmsSslProtocol);
-    return new HydratorFactory(restClient);
+    EntityClient entityClient = new EntityClient(restClient);
+    return new HydratorFactory(entityClient, SnapshotEntityRegistry.getInstance());
```
Comment (jjoyce0510) [inline metadata-jobs/mae-consumer-job/src/main/java/com/linkedin/metadata/kafka/config/HydratorFactoryConfig.java:28]: This scares me - It leaks the concept of EntityRegistry to outside of GMS. I do not feel like it should be leaked in this way if we can avoid it

File: metadata-jobs/mae-consumer-job/src/main/java/com/linkedin/metadata/kafka/hydrator/ChartHydrator.java [hunk 5]
```diff
@@ -1,45 +1,35 @@
...
 
   @Override
-  public Optional<ObjectNode> getHydratedEntity(String urn) {
-    ChartUrn chartUrn;
-    try {
-      chartUrn = ChartUrn.createFromString(urn);
-    } catch (URISyntaxException e) {
-      log.info("Invalid Chart URN: {}", urn);
-      return Optional.empty();
+  protected void hydrateFromSnapshot(ObjectNode document, Snapshot snapshot) {
```
Comment (jjoyce0510) [inline metadata-jobs/mae-consumer-job/src/main/java/com/linkedin/metadata/kafka/hydrator/ChartHydrator.java:24]: What if we typed this class on "ChartSnapshot" and actually passed in a ChartSnapshot here?

File: metadata-jobs/mae-consumer-job/build.gradle (hunk 3 above)
Comment (dexter-mh-lee) [reply metadata-jobs/mae-consumer-job/build.gradle:16]: circular dependency?

File: metadata-jobs/mae-consumer-job/src/main/java/com/linkedin/metadata/kafka/hydrator/ChartHydrator.java (hunk 5 above)
Comment (dexter-mh-lee) [reply metadata-jobs/mae-consumer-job/src/main/java/com/linkedin/metadata/kafka/hydrator/ChartHydrator.java:24]: This is the issue I talked about. If there's a way to generically cast the snapshot into chartsnapshot, yes. Otherwise, I will need another abstract function that converts snapshot into ChartSnapshot

File: metadata-jobs/mae-consumer-job/src/main/java/com/linkedin/metadata/kafka/config/HydratorFactoryConfig.java (hunk 4 above)
Comment (dexter-mh-lee) [reply metadata-jobs/mae-consumer-job/src/main/java/com/linkedin/metadata/kafka/config/HydratorFactoryConfig.java:28]: It's used to get the keyaspect schema for the entity type. Is there a way to avoid this?

**Review by jjoyce0510 — APPROVED** [review]
Looks really good! Thanks Dexter.

**Review by shirshanka — APPROVED** [review]
LGTM. Thanks @dexter-mh-lee!
